src/Microsoft.AspNetCore.OData/Formatter/Value/TypedEdmStructuredObject.cs

```diff
@@ -120,4 +121,27 @@ public bool TryGetPropertyValue(string propertyName, out object value)
             return helper.GetValue;
         }
     }
+
+    /// <summary>
+    /// A custom equality comparer for the property getter cache. 
+    /// </summary>
+    internal class PropertyGetterCacheEqualityComparer : IEqualityComparer<Tuple<string, Type>>
+    {
+        public bool Equals(Tuple<string, Type> x, Tuple<string, Type> y)
+        {
+            return x.Item1 == y.Item1 && x.Item2 == y.Item2;
+        }
+
+        public int GetHashCode(Tuple<string, Type> obj)
+        {
+            unchecked
+            {
+                int hash = 19;
+                hash = hash * 23 + obj.Item1.GetHashCode();
+                hash = hash * 23 + obj.Item2.GetHashCode();
+
+                return hash;
+            }
```

Comment:
```suggestion
            return obj.GetHashCode();
```
`Tuple` supports `GetHashCode` method such that:
```csharp
using System;

var tuple1 = Tuple.Create("typeofInt", typeof(int));
var tuple2 = Tuple.Create("typeofInt", typeof(int));
var tuple3 = Tuple.Create("typeofFlot", typeof(float));

Console.WriteLine(tuple1.GetHashCode().Equals(tuple2.GetHashCode()));
// Result: True

Console.WriteLine(tuple1.GetHashCode().Equals(tuple3.GetHashCode()));
// Result: False
```